File: solutions/c/darts/1/darts.c
#include "darts.h"

int score(coordinate_t landing_position) {
    float spot = powf(landing_position.x, 2.0) + powf(landing_position.y, 2.0);
    int points = 0;

    if ( spot <= powf(1.0, 2.0) ) {
        points = 10;
    } else if ( spot <= powf(5.0, 2.0) ) {
        points = 5;
    } else if ( spot <= powf(10.0, 2.0) ) {
        points = 1;
    }

    return points;
}

File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    spot = x * x + y * y
    cond do
      spot <=   1 -> 10
      spot <=  25 ->  5
      spot <= 100 ->  1
      true        ->  0
    end
  end
end
